spec/controllers/products_controller_spec.rb

```diff
@@ -0,0 +1,198 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+  before :each do
+    @product = Product.create(name: "starfish", price: 3, merchant_id: 1, description: "A starfish!")
+  end
+
+  let(:create_params) do
+  { merchant_id: 1,
+    product: {
+      name: "Dogfish",
+      price: 2,
+      description: "new description"
+    }
+  }
+  end
+
+  let(:update_params) do
+  { merchant_id: 1,
+    id: 1,
+    product: {
+      name: "Catfish",
+      price: 2,
+      description: "new description"
+    }
+  }
+  end
+
+  let(:bad_params) do
+    { merchant_id: 1,
+      id: 1,
+      product: {
+        name: "",
+      }
+    }
+  end
+
+  let(:merchant) do
+    Merchant.create(username: "Apple", email: "email345@email.com", password: "password", password_confirmation: "password")
+  end
+
+  let(:merchant2) do
+    Merchant.create(username: "Ricky", email: "testing@testing.com", password: "password", password_confirmation: "password")
+  end
+
+  let(:session_data) do
+    {
+      username: "Apple",
+      password: "password"
+    }
+  end
+
+  let(:good_review) do
+    {
+      id: 1,
+      review: {
+        rating: 3,
+        description: "Aight",
+        product_id: 1
+      }
+    }
+  end
+
+  describe "GET #all_products" do
+    it "is successful" do
+      get :all_products
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe "GET #index" do
+    it "is successful" do
+      merchant.authenticate(session_data)
+      session[:user_id] = merchant.id
+      get :index, merchant_id: @product.merchant_id
+      expect(response.status).to eq 200
+    end
+  end
+
+  describe "GET 'new'" do
```

Comment:
Consistency in how things are named/referenced is helpful for those consuming your code. I like your use of the pattern `describe "VERB #method"` in your specs and suggest using that here and on line 91 as well.